packages/components/forms/src/base-checkbox/BaseCheckbox.tsx

```diff
@@ -99,40 +84,43 @@ function _BaseCheckbox(
     [willBlurOnEsc, onKeyDown, inputRef],
   );
 
+  const ariaChecked =
+    typeof isChecked !== undefined ? isChecked : defaultChecked;
+
   return (
-    <Flex
-      as="label"
+    <FormLabel
       className={cx(styles.wrapper, className)}
       htmlFor={inputId}
       testId={testId}
       {...otherProps}
     >
       <input
         {...inputProps}
-        className={cx(styles.input, inputProps?.className)}
+        aria-label={ariaLabel}
+        checked={isChecked}
+        defaultChecked={defaultChecked}
+        className={styles.input}
         type={type === 'switch' ? 'checkbox' : type}
-        onChange={handleChange}
+        onChange={onChange}
         onFocus={handleFocus}
         onBlur={handleBlur}
         onKeyDown={handleKeyDown}
         value={value}
         disabled={isDisabled}
-        checked={checked}
         role={type}
-        aria-checked={isIndeterminate ? 'mixed' : checked}
+        aria-checked={isIndeterminate ? 'mixed' : ariaChecked}
         ref={inputRef}
         required={isRequired}
         id={inputId}
         name={name}
-        aria-label={label}
       />
       <GhostCheckbox
         type={type}
-        isChecked={checked}
         isDisabled={isDisabled}
         isIndeterminate={isIndeterminate}
       />
-    </Flex>
+      {children}
```

Comment:
I find it a bit less intuitive to pass a label to the component like the checkbox or radio as children. Maybe it's just me. 
I would rather expect to have something like we had before, so:
```
<Checkbox
  label="my label over here"
/>
```

But again, might be just me. 🤷‍♀️

Reply:
I think this may be just a familiarity bias, kind of how we think it's fine to have the value of the `textarea` inside the tags, and the for the `input` it has the value.